libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -131,8 +131,8 @@ class AP_BattMonitor
     bool consumed_wh(float&wh, const uint8_t instance = AP_BATT_PRIMARY_INSTANCE) const WARN_IF_UNUSED;
 
     /// capacity_remaining_pct - returns the % battery capacity remaining (0 ~ 100)
-    virtual uint8_t capacity_remaining_pct(uint8_t instance) const;
-    uint8_t capacity_remaining_pct() const { return capacity_remaining_pct(AP_BATT_PRIMARY_INSTANCE); }
+    virtual bool capacity_remaining_pct(int8_t &percentage, uint8_t instance) const WARN_IF_UNUSED; 
```

Comment:
Contrary to a lot of C function calls, we wouldn't usually put the return parameter first here.

By convention if a method takes an instance number - that's the first in the list.

```suggestion
    virtual bool capacity_remaining_pct(uint8_t instance, int8_t &percentage) const WARN_IF_UNUSED; 
```

I'm not keen on passing in `int8_t` vs `uint8_t` - but that's not your fight.

Reply:
The three functions above this take instance as the second argument, should I still make this one take it as first argument anyway?

Reply:
@peterbarker we are trying to get this into a state where it may be merged on the call tonight. What should we do here?
- apply your suggested change, and leave the surrounding code as is
- keep willian's original patch to keep the signatures in this file consistent (but against your convention)
- apply your suggested change to the new signature, as well as the surrounding signatures in order to make all the signatures in this file match your convention (this will touch a lot of code)